nbbuild/build.xml

```diff
@@ -1946,30 +1946,41 @@ It is possible to use -Ddebug.port=3234 -Ddebug.pause=y to start the system in d
+            <exclude name="diff/test/unit/src/org/netbeans/modules/diff/builtin/visualizer/data/**" /> <!--test data-->
+            <exclude name="utilities/test/unit/src/org/netbeans/modules/openfile/resources/recent_files/*" /> <!--test data-->
+            <exclude name="**/*.url" /> <!--no degree of creativity-->
+        </patternset>
         <rat:report xmlns:rat="antlib:org.apache.rat.anttasks" reportfile="build/rat-report.txt">
             <fileset dir="${nb_all}">
-                <exclude name="*/build/**" />
-                <exclude name="nbbuild/netbeans/**" />
-                <exclude name="**/manifest.mf" /> <!--do not nativelly support comments-->
-                <exclude name="*/nbproject/*.sig" /> <!--generated signatures for past versions-->
-                <exclude name="*/nbproject/build-impl.xml" /> <!--generated, no degree of creativity -->
-                <exclude name="*/nbproject/jfx-impl.xml" /> <!--generated, no degree of creativity -->
-                <exclude name="*/nbproject/genfiles.properties" /> <!--generated, no degree of creativity -->
-                <exclude name="*/nbproject/private/**" /> <!--user-specific files -->
-                <exclude name="*/external/*-license.txt" /> <!--licenses for external dependencies-->
-                <exclude name="*/external/*-notice.txt" /> <!--notices for external dependencies-->
-                <exclude name="**/*.pass" /> <!--generated test files-->
-                <exclude name="**/*.pass2" /> <!--generated test files-->
-                <exclude name="**/.list" /> <!--no degree of creativity-->
-                <exclude name="**/src/META-INF/**" /> <!--no degree of creativity-->
-                <exclude name="form/test/unit/data/goldenfiles/*" /> <!--test data-->
-                <exclude name="diff/test/unit/data/**" /> <!--test data-->
-                <exclude name="diff/test/unit/src/org/netbeans/modules/diff/builtin/provider/*.txt" /> <!--test data-->
-                <exclude name="diff/test/unit/src/org/netbeans/modules/diff/builtin/visualizer/data/**" /> <!--test data-->
-                <exclude name="utilities/test/unit/src/org/netbeans/modules/openfile/resources/recent_files/*" /> <!--test data-->
-                <exclude name="**/*.url" /> <!--no degree of creativity-->
+                <patternset refid="non.rat" />
+            </fileset>
+        </rat:report>
+        <rat:report xmlns:rat="antlib:org.apache.rat.anttasks" format="xml" reportfile="build/rat-report.xml">
+            <fileset dir="${nb_all}">
+                <patternset refid="non.rat" />
             </fileset>
         </rat:report>
+        <taskdef name="rattohtml" classname="org.netbeans.nbbuild.reporting.RatHtmlReportTask" classpath="${build.ant.classes.dir}"/>
+        <loadproperties srcFile="cluster.properties" />
+        <rattohtml source="build/rat-report.xml" target="build/rat-report.html"/>
```

Comment:
I am not sure whether HTML is that much better format than txt. Probably the direct hyperlinks to the Git repo are the biggest improvement, right?

In any case I was looking for a way to track progress of our rat related efforts. Just instead of HTML it should generate JUnit XML report files. If the output used [JUnitReportWritter](https://github.com/ebarboni/incubator-netbeans/blob/a343abdcc6026022f60792a2905aa3da6eb95853/nbbuild/antsrc/org/netbeans/nbbuild/JUnitReportWriter.java) just like the other classes in the `nbbuild/antsrc`, we could set the continuous job to show us how many violations we still have and track the progress.

Reply:
Sorry but the last commit with refactoring kill the converstation flow.

Maybe hyperlink is for now the only improvement for having HTML report. By cluster information can be done in txt file.

Helping tooling continuous job is a big point. But we should also think on people who will review and vote for release.
for example http://commons.apache.org/proper/commons-bcel/  has a project information section including a rat report.